src/Domain/ValueObjects/WalletValueObject.cs

```diff
@@ -0,0 +1,43 @@
+using Domain.Aggregates;
+
+namespace Domain.ValueObjects;
+
+public record Wallet(decimal Balance)
+{
+    public User User { get; set; }
```

Comment:
no need for user here, a wallet is just a wrapper object, we will still save the decimal value in the database. wallet only exists on a user, in memory, not in the database